cmd/setup-etcd-environment/run_test.go

```diff
@@ -0,0 +1,133 @@
+    m := make(map[string]string)
+    m["INITIAL_CLUSTER"] = initialClusterStr;
+    m["INITIAL_CLUSTER_STATE"] = "existing"
+
+    writeEnvironmentFile(m, file, false)
+    
+    bootEnv, err := setBootstrapEnv(file.Name(), "etcdname.devcluster.openshift.com", true)
+
+    if err != nil {
+        t.Fatalf("setBootstrapEnv() failed: %s", err)
+    }
+
+    if bootEnv == nil { 
+        t.Errorf("bootEnv is unexpectedly nil")
+    }
+
+    if bootEnv["DISCOVERY_SRV"] != "" {
+        t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = \"%s\"; want \"\" ", bootEnv["DISCOVERY_SRV"])
+    }
+
+    if bootEnv["INITIAL_CLUSTER"] != initialClusterStr {
+        t.Errorf("bootEnv[\"INITIAL_CLUSTER\"] = \"%s\"; want \"%s\" ", bootEnv["INITIAL_CLUSTER"], initialClusterStr)
+    }
+    if bootEnv["INITIAL_CLUSTER_STATE"] != "existing" {
+        t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = \"%s\"; want \"existing\" ", bootEnv["INITIAL_CLUSTER_STATE"])
+    }
+}
+
+func TestFileWithDiscoverySrv(t *testing.T) {
+    file, err := ioutil.TempFile("/tmp/", "etcd_")
+    if err != nil {
+       t.Fatalf("TestFileWithInitialCluster failed: %s", err)
+    }
+    defer os.Remove(file.Name())
+    m := make(map[string]string)
+    m["DISCOVERY_SRV"] = "OLDNAME.devcluster.openshift.com";
+
+    writeEnvironmentFile(m, file, false)
+    
+    bootEnv, err := setBootstrapEnv(file.Name(), "etcdname.devcluster.openshift.com", true)
+
+    if err != nil {
+        t.Fatalf("setBootstrapEnv() failed: %s", err)
+    }
+
+    if bootEnv == nil { 
+        t.Errorf("bootEnv is unexpectedly nil")
+    }
+
+    if bootEnv == nil || bootEnv["DISCOVERY_SRV"] != "etcdname.devcluster.openshift.com" {
+        t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = \"%s\"; want \"etcdname.devcluster.openshift.com\" ", bootEnv["DISCOVERY_SRV"])
+    }
+
+    if bootEnv["INITIAL_CLUSTER"] != "" {
+        t.Errorf("bootEnv[\"INITIAL_CLUSTER\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER"])
+    }
+    if bootEnv["INITIAL_CLUSTER_STATE"] != "" {
+        t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER_STATE"])
+    }
+}
+
+func TestFileWithNoVariables(t *testing.T) {
+    file, err := ioutil.TempFile("/tmp/", "etcd_")
+    if err != nil {
+       t.Fatalf("TestFileWithNoVariables failed: %s", err)
+    }
+    defer os.Remove(file.Name())
+
+    bootEnv, err := setBootstrapEnv(file.Name(), "etcdname.devcluster.openshift.com", true)
+
+    if err != nil {
+        t.Fatalf("setBootstrapEnv() failed: %s", err)
+    }
+
+    if bootEnv == nil || bootEnv["DISCOVERY_SRV"] != "etcdname.devcluster.openshift.com" {
+        t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = \"%s\"; want \"etcdname.devcluster.openshift.com\" ", bootEnv["DISCOVERY_SRV"])
+    }
+    if bootEnv["INITIAL_CLUSTER"] != "" {
+        t.Errorf("bootEnv[\"INITIAL_CLUSTER\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER"])
```

Comment:
curious why the choice of Error vs Fatal in general

Reply:
Fatalf = FailNow() + Logf()
Errorf = Fail() + Logf()

FailNow() fails the function and returns from the function after printing the log.
Fail() marks the function failure and logs, but continues the execution of the function.